src/rust/ide/json-rpc/src/macros.rs

```diff
@@ -202,6 +220,10 @@ macro_rules! make_rpc_methods {
 /// ```
 #[macro_export]
 macro_rules! expect_call {
+    ($mock:ident.$method:ident() => $result:expr) => {
+        let result = $result;
+        $mock.expect.$method(move || result)
+    };
```

Comment:
The next branch would lead to infinite recursion in this case. 
The next branch purpose is to replace `(arg,…)` with `(arg=val,…)`. However, when there are no arguments, it becomes self-entrant no-op and there is no means to enter the third branch.